### PR TITLE
!feat: proposal creation helpers

### PR DESCRIPTION
**Pull request by sakulstra**
BREAKING:
- GovHelpers no longer exports any constant addresses
- `GovHelpers.createProposal` was renamed to `GovHelpers.createProposalAsWhale`

### CODE REVIEW

File: src/GovHelpers.sol [hunk 1]
```diff
@@ -8,6 +8,86 @@ import {AaveGovernanceV2, IAaveGovernanceV2, IExecutorWithTimelock} from 'aave-a
 import {IPoolAddressesProvider} from 'aave-address-book/AaveV3.sol';
 import {ProxyHelpers} from './ProxyHelpers.sol';
 
+library DelegatecallProposalCreationHelper {
```
Comment (sakulstra) [inline src/GovHelpers.sol]: Should this even be a special library or part of `GovHelpers` ?

File: src/GovHelpers.sol [hunk 2]
```diff
@@ -8,6 +8,86 @@ import {AaveGovernanceV2, IAaveGovernanceV2, IExecutorWithTimelock} from 'aave-a
...
+  struct DelegateCallProposal {
+    address target;
+    string signature;
+    bytes callData;
+  }
+
+  function createMainnetDelegateCall(address payloadAddress)
```
Comment (sakulstra) [inline src/GovHelpers.sol]: not sure about the naming

File: src/GovHelpers.sol [hunk 3]
```diff
@@ -8,6 +8,86 @@ import {AaveGovernanceV2, IAaveGovernanceV2, IExecutorWithTimelock} from 'aave-a
...
+  {
+    return
+      DelegateCallProposal({
+        target: AaveGovernanceV2.CROSSCHAIN_FORWARDER_POLYGON,
+        signature: 'execute(address)',
+        callData: abi.encode(payloadAddress)
+      });
+  }
+
+  function createProposal(DelegateCallProposal[] memory delegateCalls, bytes32 ipfsHash)
```
Comment (sakulstra) [inline src/GovHelpers.sol]: we already have `createProposal` in `GovHelpers` - the main difference beeing that it automatically mocks a whale etc, so it's not usable in breadcast. Does it make sense to rename the mock one to `createProposalViaPrankedWhale` or similar?

File: src/GovHelpers.sol (hunk 2 above)
Comment (eboadom) [reply src/GovHelpers.sol]: Why not `createMainnet()`, you already have the other on the Lib name

**Review by sendra — APPROVED** [review]

File: src/GovHelpers.sol [hunk 4]
```diff
@@ -19,24 +20,93 @@ library GovHelpers {
...
+      targets[i] = delegateCalls[i].target;
+      signatures[i] = delegateCalls[i].signature;
+      calldatas[i] = delegateCalls[i].callData;
+      values[i] = 0;
+      withDelegatecalls[i] = true;
+    }
+
+    return
+      AaveGovernanceV2.GOV.create(
+        IExecutorWithTimelock(AaveGovernanceV2.SHORT_EXECUTOR),
```
Comment (eboadom) [inline src/GovHelpers.sol:96]: We should remember that this is biased for short exec and not screw on some long exec proposal

Comment (sakulstra) [reply src/GovHelpers.sol:96]: we might add an overload allowing to pass an executor

**Pull request merged**